### PR TITLE
Use dynamic returns for ray_execute_bundle

### PR DESCRIPTION
**Pull request by jjyao**
Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>

With dynamic returns feature, we no longer need to calculate the number of returns beforehand.

### CODE REVIEW

**Comment by jjyao** [comment]
GH has some issue right now so CI is not triggered. Don't merge it until I run all tests.

**Review by iasoon — APPROVED** [review]
Looks great! More functionality, and the code also looks nicer than before imo.
Can be merged for me once the tests have run, thanks @jjyao !

**Comment by pabloem** [comment]
this is awesome. Thanks LGTM!

File: ray_beam_runner/portability/execution.py [hunk 1]
```diff
@@ -144,22 +142,13 @@ def ray_execute_bundle(
...
-    )
+    returns.append(len(delayed_applications))
     for pcoll, buffer in delayed_applications.items():
-        returns = returns + (pcoll, buffer)
-    # Caller expects num_input_transforms number of delayed applications,
-    # so fill the remaining with Nones.
-    for _ in range(len(delayed_applications), num_input_transforms):
-        returns = returns + (None, None)
+        returns.append(pcoll)
+        returns.append(buffer)
```
[the remark this answers is not in the document]
Comment (jjyao) [reply ray_beam_runner/portability/execution.py:148]: Yes, we can. I saved to a list first for easy debugging (so I can print out all the returns).

**Pull request merged**